### PR TITLE
(3021) Exclude health check requests from host auth

### PR DESCRIPTION
**Pull request by CristinaRO**
## Changes in this PR

Exclude health check requests from host authorisation, to allow the AWS deploy pipeline to proceed faster. Currently health check requests are denied by the host auth process, because the IPs are not in the DOMAIN or ADDITIONAL_HOSTNAMES env vars.

We don't have a canonical list of the IPs, nor do we have easy access to the env vars (these particular env vars live in task definitions, not in ParameterStore).

https://api.rubyonrails.org/classes/ActionDispatch/HostAuthorization.html

## Screenshots of UI changes

N/A

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by CristinaRO** [comment]
[Get rid of unauthorised host error for AWS health check requests](https://trello.com/c/wGtEwO3O/3021-get-rid-of-unauthorised-host-error-for-aws-health-check-requests)

File: config/application.rb [hunk 1]
```diff
@@ -74,5 +74,7 @@ class Application < Rails::Application
 
     # Default headers
     config.action_dispatch.default_headers["X-XSS-Protection"] = "0"
+
+    config.host_authorization = {exclude: ->(request) { request.path =~ /health_check/ }}
```
Comment (mec) [inline config/application.rb:78]: Do we need to include the `/health-check/` path as well?

**Review by mec — CHANGES_REQUESTED** [review]

File: config/application.rb (hunk 1 above)
Comment (CristinaRO) [reply config/application.rb:78]: I was counting on the fact that the app responds to both health-check and health_check requests, but we might, not sure.

Comment (CristinaRO) [reply config/application.rb]: I think you're right, the request needs to match, even if it's processed by the same controller. PR updated.

**Review by mec — APPROVED** [review]
🚢

**Pull request merged**